### PR TITLE
Save PeerScout API recommendations to BQ table

### PR DESCRIPTION
**Pull request by HazalCiplak**
https://github.com/elifesciences/data-hub-issues/issues/443

### CODE REVIEW

File: peerscout_api/main.py [hunk 1]
```diff
@@ -356,6 +369,28 @@ def get_response_json(
...
+def write_peerscout_api_response_to_bq(
+    recommendation_request: dict,
+    recommendation_response: dict
+):
+    PROJECT_NAME = 'elife-data-pipeline'
+    DATASET_NAME = get_deployment_env()
+    recommendation_response_with_provenance = remove_key_with_null_value({
+        **recommendation_response,
+        'provenance': {
+            **recommendation_request,
```
Comment (de-code) [inline peerscout_api/main.py:381]: maybe it would be good to keep it as a `request` field to make it clear where it is coming from?

**Comment by HazalCiplak** [comment]
> I think when we were just reading, it was fine to use `staging`. But we might want to avoid writing to `staging` as part of the `ci` end to end tests, or running it locally.

Then we need to create an additional env variable for this. How does `TARGET_DEPLOYMENT_ENV_VALUE_FOR_TEST_API` sound?

**Comment by de-code** [comment]
> > I think when we were just reading, it was fine to use `staging`. But we might want to avoid writing to `staging` as part of the `ci` end to end tests, or running it locally.
> 
> Then we need to create an additional env variable for this. How does `TARGET_DEPLOYMENT_ENV_VALUE_FOR_TEST_API` sound?

Is it then only used for test?

Maybe use something like `PEERSCOUT_API_TARGET_DATASET`?

File: peerscout_api/main.py [hunk 2]
```diff
@@ -356,6 +379,29 @@ def get_response_json(
         }
 
 
+def write_peerscout_api_response_to_bq(
+    recommendation_request: dict,
+    recommendation_response: dict
+):
+    PROJECT_NAME = 'elife-data-pipeline'
+    TARGET_DATASET_NAME = get_target_dataset_env()
```
Comment (de-code) [inline peerscout_api/main.py:387]: would suggest to use lower case for any local variables and constants only at the module level?

File: tests_peerscout_api/unit_tests/main_test.py [hunk 3]
```diff
@@ -174,6 +184,11 @@ def _get_ok_json(response):
     return response.json
 
 
+class TestGetTargetDatasetEnv:
+    def test_should_read_target_dataset_as_ci_for_e2e_test(self):
```
Comment (de-code) [inline tests_peerscout_api/unit_tests/main_test.py:188]: I believe this is not directly related to end to end tests. Ideally the unit test wouldn't depend on environment variables set outside..

In Data Hub Core DAGs there is a `mock_env` fixture defined: https://github.com/elifesciences/data-hub-core-airflow-dags/blob/develop/tests/unit_test/conftest.py#L22-L25

You could copy it into `conftest.py` in this repo too and use it here like:

```python
@pytest.mark.usefixtures('mock_env')
def test_should_return_ci_by_default(self):
    assert get_target_dataset_env() == 'ci'
    # this could also refer to DEFAULT_PEERSCOUT_API_TARGET_DATASET_VALUE (in which case the test name would be changed slightly)


def test_should_return_value_of_peerscout_api_target_dataset_env_var(self, mock_env: dict):
    mock_env['PEERSCOUT_API_TARGET_DATASET'] = 'dataset1'
    assert get_target_dataset_env() == 'dataset1'

```

**Review by de-code — APPROVED** [review]
